### PR TITLE
Discover and Get commands for Source Registry

### PR DESCRIPTION
**Pull request by VladimirZaets**
## Description
This pull request adds new commands to allow the developer to work with the api-mesh sources catalog through CLI.

API mesh Source is an entity that declares GraphQL schema and maps it to 3rd party APIs or Runtime Actions.API Mesh config that can be meshed into the user's Mesh schema.

The source is an entity that declares GraphQL schema and maps it to 3rd party APIs or Runtime Actions. GraphQL Mesh config that can be meshed into the user's Mesh schema. To learn more about how GraphQL API Gateway manages configs.
Motivation and Context

## Motivation and Context
To learn more about api-mesh sources functionality, please check this wiki document. https://wiki.corp.adobe.com/display/dxdevx/API+mesh+connector%28sources%29+Customer+Experience

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/api-mesh/source/get.js
Comment (roribio) [inline src/commands/api-mesh/source/get.js:33]:
```suggestion
				`The "aio api-mesh:source:get" command requires additional parameters` +
```

File: src/commands/api-mesh/source/get.js
Comment (roribio) [inline src/commands/api-mesh/source/get.js:113]:
```suggestion
GetCommand.description = 'Command returns the content of a specific connector';
```

**Review by roribio — APPROVED** [review]
A few typos, otherwise looks good 👍

**Comment by of0x** [comment]
Can we please have a general description of the feature and a few sentences for Motivation and Context section?

**Comment by irenelagno** [comment]
@of0x, I can't modify the description of the pull request. So, I added info in the comment.
**Description**
This pull request adds new commands to allow the developer to work with the api-mesh sources catalog through CLI. 

API mesh Source is an entity that declares GraphQL schema and maps it to 3rd party APIs or Runtime Actions.API Mesh config that can be meshed into the user's Mesh schema. 

The source is an entity that declares GraphQL schema and maps it to 3rd party APIs or Runtime Actions. GraphQL Mesh config that can be meshed into the user's Mesh schema. To learn more about how GraphQL API Gateway manages configs.
**Motivation and Context**
To learn more about api-mesh sources functionality, please check this wiki document. https://wiki.corp.adobe.com/display/dxdevx/API+mesh+connector%28sources%29+Customer+Experience


cc @VladimirZaets please update the pull request description.

File: src/commands/api-mesh/source/__fixtures__/0.0.1-test-01.json
Comment (revanth0212) [inline src/commands/api-mesh/source/__fixtures__/0.0.1-test-01.json:43]: I understand this is just a test but we don't support cache transform at the moment. If this is submitted to SMS, it will fail with an appropriate error.

**Review by revanth0212 — CHANGES_REQUESTED** [review]
Nice work guys. I have some feedback, a couple of them are just suggestions in case you want to consider them.

I am happy to elaborate more on them during our call this afternoon.

File: src/commands/api-mesh/source/__tests__/discover.test.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/__tests__/discover.test.js:35]: I add base snapshots for the commands, but in other cases I tests same data like in snapshot but using fixtures (because output is pretty large).

File: src/commands/api-mesh/source/get.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/get.js:1]: not now, currently we have only one environment, moreover, CLI it's just a client, the URL for the source storage can manage prod and stage env

File: src/commands/api-mesh/source/get.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/get.js:114]: discussed it on the meeting, not relevant anymore.

File: package.json
Comment (VladimirZaets) [reply package.json]: Installed node-clipboard instead. The same with clipboard but adapted to use in nodejs

File: src/commands/api-mesh/source/discover.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/discover.js]: Wrapped remote call

Comment (VladimirZaets) [reply src/commands/api-mesh/source/discover.js]: done

File: src/commands/api-mesh/source/get.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/get.js]: done

File: src/commands/api-mesh/source/__tests__/get.test.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/__tests__/get.test.js:50]: I add base snapshots for the commands, but in other cases I tests same data like in snapshot but using fixtures (because output is pretty large).

File: src/commands/api-mesh/source/__fixtures__/0.0.1-test-01.json
Comment (VladimirZaets) [reply src/commands/api-mesh/source/__fixtures__/0.0.1-test-01.json]: Removed

File: src/commands/api-mesh/source/get.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/get.js]: done

File: src/commands/api-mesh/source/get.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/get.js]: done

File: src/commands/api-mesh/source/discover.js
Comment (VladimirZaets) [reply src/commands/api-mesh/source/discover.js:1]: done

File: package.json
Comment (VladimirZaets) [reply package.json]: It was my mistake, reverted it back. But just FYI. Nodejs 8 is unsupported from 2020, the least LTS version is 14 right now. As I know Adobe provide requirements to update nodejs version

**Comment by VladimirZaets** [comment]
@revanth0212 thank you for review, I updated all your suggestions, please look it again. Thanks!

File: package.json
Comment (revanth0212) [reply package.json]: That's good to know. Ill create a ticket in our backlog to upgrade the minimum version.

File: src/commands/api-mesh/source/get.js
Comment (revanth0212) [inline src/commands/api-mesh/source/get.js:30]: Please wrap the whole run function in a try-catch. If something in the function fails, the customer will be left with error stack that might not make sense to them. Please wrap it in a try-catch and add logger.error inside so we can get error stack information if needed while debugging.

File: README.md
Comment (revanth0212) [inline README.md:83]: Can you add a brief description of the source registry so the customers can have some context while checking the readme?

File: README.md
Comment (revanth0212) [inline README.md:85]:
```suggestion
```

File: src/commands/api-mesh/source/discover.js
Comment (revanth0212) [inline src/commands/api-mesh/source/discover.js:22]: Same here. Please wrap the whole function in a try-catch block.

**Review by revanth0212 — CHANGES_REQUESTED** [review]
Nice work @VladimirZaets 

Almost there. I just have 2 more comments and we should be good to go.

**Comment by VladimirZaets** [comment]
@revanth0212 done! Thank you for review, please check my changes again!

File: README.md
Comment (revanth0212) [inline README.md:87]:
```suggestion
Source registry is a collection of predefined sources (API mesh source configurations) that are created to solve specific use cases. The source can be installed for customer-specific API mesh configuration. 

To submit a new source, please follow the instructions provided in the [Source Registry](https://github.com/adobe/api-mesh-sources/) repository.
```

**Review by revanth0212 — APPROVED** [review]
Looks good @VladimirZaets 

Thank you for making the requested changes. Please verify the functionality before merging (if not done already).

**Review by revanth0212 — APPROVED** [review]

**Comment by revanth0212** [comment]
Ready to Merge once functionality validation is confirmed

**Pull request merged**
